Testing!

edomora97 opened the pull request:
:tada: Added a test suite! :tada:

edomora97 committed:
Added test suite

Added Config tests

edomora97 committed:
Added more tests

edomora97 committed:
Added tests for Logger

edomora97 committed:
Added tests for StorageManager

Added util to supress stderr messages

edomora97 committed:
Added travis-ci support!

edomora97 committed:
Added isolation_level=None to the database connection

edomora97 committed:
Trying to update pypy to the latest version

edomora97 committed:
Implemented tests for AdminHandeler and fixed stuff in Database

edomora97 committed:
It seems that assert_called_once is for py >= 3.6

edomora97 committed:
Added tests for BaseHandler

edomora97 committed:
Implemented tests for ContestHandler

edomora97 committed:
Patched tempfile to use a different tmp directory

edomora97 committed:
Reimplemented last commit in an better way

edomora97 committed:
Added coveralls.io support

edomora97 committed:
Dho! I've written a "little" typo

edomora97 committed:
Added badges and fixed a small problem

edomora97 committed:
Added tests for InfoHandler

edomora97 committed:
Added tests for UploadHandler

edomora97 committed:
Added a lot of tests

edomora97 committed:
Fixed a test

edomora97 committed:
Added tests for ContestManager

edomora97 committed:
Fixed some tests

edomora97 committed:
Small bufix in a test

edomora97 committed:
WIP: Added a small test for server

edomora97 committed:
Completed implementation of TestServer

edomora97 committed:
Small fix

edomora97 committed:
Fixed test and added missing one

edomora97 committed:
Torn down temp files

edomora97 committed:
Replaced assertTrue(str.find(str) >= 0) with assertIn

edomora97 committed:
Moved a temp file

edomora97 committed:
Implemented skipped test in AdminHandler

edomora97 committed:
Fixed problem with the incoming merge

wil93 commented on run_tests.py:
```diff
@@ -0,0 +1,28 @@
+# trying to use a subdirectory of /tmp (verytmp)
+import tempfile
+tmpDirs = tempfile._candidate_tempdir_list()
+for dir in tmpDirs:
```
The `dir` name could be changed to something else which is not a built-in function 😄

edomora97 replied:
I think in this crappy piece of code `dir` is the last of the problems :joy:

Before merging to `master` I may try to find an other way to manage temporary files. These 10 lines of code are here only because after a day running the tests my `/tmp` folder was full of junk (over 50k files!).

Thing about reimplementing `tempfile` module...

edomora97 committed:
Customized temp file/folder generation

veluca93 commented on src/contest_manager.py:
```diff
@@ -48,10 +49,9 @@ def import_contest(path):
-                shutil.copytree(os.path.join(taskdir, "statement"), statementdir)
+            if os.path.isdir(statementdir):
+                shutil.rmtree(statementdir)
+            shutil.copytree(os.path.join(taskdir, "statement"), statementdir)
```
What happens now if statementdir is a file and not a directory?

edomora97 replied:
Something bad...

Before there was an other error, note that `shutil.rmtree` can only remove directories!
An error is raised if you try to remove a file
```NotADirectoryError: [Errno 20] Not a directory: '/tmp/foobar.txt'```

If `statementdir` is a file we have to remove it. The question now is: if it's a directory should we keep it?

veluca93 replied:
I don't think so

veluca93 commented on test/handlers/test_admin_handler.py:
```diff
@@ -0,0 +1,223 @@
+        self.assertEqual("token2", user2["token"])
+        self.assertEqual(0, len(user2["ip"]))
+
+    def _preapre_zip(self):
```
s/preapre/prepare/ ?

edomora97 replied:
ops :joy:

veluca93 commented on test/handlers/test_base_handler.py:
```diff
@@ -0,0 +1,253 @@
+    def test_get_ip_3_proxies(self):
+        Config.num_proxies = 3
+        headers = { "X-Forwarded-For": "1.2.3.4, 5.6.7.8, 8.8.8.8" }
+        request = Request(EnvironBuilder(headers=headers).get_environ())
```
Shouldn't we also add a REMOTE_ADDR parameter?

edomora97 replied:
K, even if it shouldn't make any differences

veluca93 commented on test/utils.py:
```diff
@@ -0,0 +1,90 @@
+    def get_tmp_dir():
+        if Utils.prefix is None:
+            Utils.prefix = os.path.join(tempfile.gettempdir(), "territoriali-backend-tests",
+                                        datetime.datetime.now().strftime("temp-%Y-%m-%d_%H-%M-%S"))
```
What is the point of concatenating the result of gettempdir with the current time?

edomora97 replied:
Each run of `./run_tests.py` has it's temp files in an isolated folder. You should be able to debug each run a bit better... Also you won't have a directory in you fs with thousands of files.

`gettempdir` should point to a valid temporary directory, the current time acts like an id for the run of the tests

veluca93 replied:
Ok, makes sense.

edomora97 committed:
Fixes

- Removed task statement dir if it's a file
- Typo
- Modified a test with REMOTE_ADDR missing
